diff --git a/src/NextServiceProvider.php b/src/NextServiceProvider.php
index 4c1ee60..e36e933 100644
--- a/src/NextServiceProvider.php
+++ b/src/NextServiceProvider.php
@@ -16,11 +16,11 @@ class NextServiceProvider implements ServiceModifierInterface {
    * {@inheritdoc}
    */
   public function alter(ContainerBuilder $container) {
-    $cors_config = $container->getParameter('cors.config');
-    if (!$cors_config['enabled']) {
-      $cors_config['enabled'] = TRUE;
-      $container->setParameter('cors.config', $cors_config);
-    }
+//    $cors_config = $container->getParameter('cors.config');
+//    if (!$cors_config['enabled']) {
+//      $cors_config['enabled'] = TRUE;
+//      $container->setParameter('cors.config', $cors_config);
+//    }
   }
 
 }

